scylla/src/frame/types.rs

```diff
@@ -40,6 +40,15 @@ impl std::fmt::Display for Consistency {
     }
 }
 
+impl Consistency {
+    pub fn validate_serial(cl: &Option<Self>) {
+        match cl {
+            None | Some(Consistency::Serial) | Some(Consistency::LocalSerial) => (),
+            _ => panic!("Serial consistency can only be set to Serial or LocalSerial"),
```

Comment:
By the way, I wonder if we couldn't split the `Consistency` type into a regular `Consistency` + `SerialConsistency` - there would be no need for runtime validation and less confusion for the users. Does it make sense to set `Serial` or `LocalSerial` for regular consistency?

On the other hand other drivers always seem to use the same consistency level type both for regular and serial consistency, maybe there is a good reason for that...

Reply:
That would indeed be best, but that's also why I mentioned maintaining backward compatibility - existing code could break for some users. That's why I went with a runtime check.

Reply:
Sigh, on the other hand, we're still in 0.x, so we can make breaking changes, and it makes so much sense to split the types... they are absolutely mutually exclusive, as it's also not legal to set a LOCAL_SERIAL level to the regular consistency level. I'll rework this pull request to make this breaking change and send it for review.

Reply:
There's also no point in following other drivers in that matter - if anything, our solution would just be more comprehensible to the users. It's ultimately CQL's fault that these levels were standardized in such a weird manner.

Reply:
@kostja Could you comment on https://github.com/scylladb/scylla-rust-driver/pull/291#discussion_r710885762 ? The rust driver has a type which represents the CQL's `[consistency]`, but would like to split it into two - one which has SERIAL and LOCAL_SERIAL consistency levels, and one which has the others. Is there a good reason _not_ to do that? I don't think any other driver which I know does that, but I'm not sure if there is a good reason behind that.

Reply:
I would be wonderful if there are two distinct types.

Reply:
SERIAL and LOCAL-SERIAL should be part of Consistency data type, but it's best to make SerialConsistency a distinct type to avoid programming bugs.